test/test_cwp.py

```diff
@@ -0,0 +1,52 @@
+# type: ignore
+from bpmncwpverify.cwp import Cwp
+from bpmncwpverify.state import SymbolTable
+from returns.pipeline import is_successful
+import os
+
+current_directory = os.path.dirname(os.path.abspath(__file__))
+
+cwp_path = os.path.join(current_directory, "example", "cwp.xml")
+
+
```

Comment:
I would write more tests, and specifically, I would write tests that separate out the `from_xml` and the actual LTL generation.  I would also write tests for anything that generates "Promela". What that means is that you instantiate, in the test, the `Cwp` instance that only has _what_ you want to test. Then you use the visitor to make sure that the Promela string it generates matches the expected string. No files. You can also further isolate the tests to test specific parts of the visitor such as `write_exists`, `write_state`, `write_log`, etc. There should be a test for each of these functions to be sure that in isolation, they do exactly what is expected. The same is true for the Promela generation. We need the break down the functionality into smaller pieces and make sure each piece is behaving as expected. And then have a few full scale integration tests as you do here.